From ef409d1af1ed747aa3b4d483ee45bedf90d0eb2e Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Tue, 14 May 2024 12:55:56 -0700
Subject: [PATCH 4/5] Add memset and memcpy weak attr

---
 lib/init.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/init.c b/lib/init.c
index 726e493..cbdb1dd 100644
--- a/lib/init.c
+++ b/lib/init.c
@@ -187,7 +187,7 @@ EFIDebugVariable (
 #define __SIZE_TYPE__ UINTN
 #endif
 
-void *memset(void *s, int c, __SIZE_TYPE__ n)
+ __attribute__ ((weak)) void *memset(void *s, int c, __SIZE_TYPE__ n)
 {
     unsigned char *p = s;
 
@@ -197,7 +197,7 @@ void *memset(void *s, int c, __SIZE_TYPE__ n)
     return s;
 }
 
-void *memcpy(void *dest, const void *src, __SIZE_TYPE__ n)
+ __attribute__ ((weak)) void *memcpy(void *dest, const void *src, __SIZE_TYPE__ n)
 {
     const unsigned char *q = src;
     unsigned char *p = dest;
-- 
2.45.0

